File: android/src/main/kotlin/com/edfapg/flutter/sdk/eventhandlers/SadadPayEventHandler.kt
package com.edfapg.flutter.sdk.eventhandlers

import android.content.Context
import com.edfapg.flutter.sdk.helper.toMap
import com.edfapg.sdk.EdfaSadadPay
import com.google.gson.Gson
import io.flutter.plugin.common.EventChannel

class SadadPayEventHandler(private val context: Context): EventChannel.StreamHandler {
    var sink:EventChannel.EventSink? = null
    private val gson = Gson()

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        sink = events
        (arguments as? Map<*, *>)?.let {
            with(it) {
                val orderId = (get("orderId") as? String)
                val orderDescription = (get("orderDescription") as? String)
                val amount = (get("amount") as? Double) ?: 0.0
                val customerName = (get("customerName") as? String)
                val mobileNumber = (get("mobileNumber") as? String)

                payWithSadad(orderId, orderDescription,amount, customerName, mobileNumber)
            }
        }
    }

    override fun onCancel(arguments: Any?) {

    }


    fun payWithSadad(orderId: String?, orderDescription: String?, amount: Double, customerName: String?, mobileNumber: String?){
        EdfaSadadPay()
            .setOrderId(orderId ?: "")
            .setOrderDescription(orderDescription ?: "")
            .setOrderAmount(amount)
            .setCustomerName(customerName ?: "")
            .setMobileNumber(mobileNumber ?: "")
            .onFailure { errors, exception ->
                print("${errors?.message} $exception")
                sink?.success(mapOf("error" to errors?.toMap()))
            }.onSuccess { res ->
                sink?.success(mapOf("success" to res.toMap()))
            }.initialize(
                onError = {
                    sink?.success(mapOf("error" to it))
                }
            )
    }
}

File: ios/Classes/EventHandlers/SadadPayEventHandler.swift

//
//  SaleEventHandler.swift
//  edfapay_sdk
//
//  Created by Zohaib Kambrani on 03/03/2023.
//

import Foundation
import Flutter
import UIKit
import EdfaPgSdk 


class SadadPayEventHandler : NSObject, FlutterStreamHandler{
    
    var eventSink:FlutterEventSink? = nil
    
    public func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        eventSink = events
        debugPrint("native.SadadPayEventHandler.onListen")

        if let params = arguments as? [String:Any]{
            
            let orderId = (params["orderId"] as? String)
            let orderDescription = (params["orderDescription"] as? String)
            let amount = (params["amount"] as? Double) ?? 0.0
            let customerName = (params["customerName"] as? String)
            let mobileNumber = (params["mobileNumber"] as? String)
            EdfaSadadPay()
                .setOrderId(orderId ?? "")
                .setOrderDescription(orderDescription ?? "")
                .setOrderAmount(amount)
                .setCustomerName(customerName ?? "")
                .setMobileNumber(mobileNumber ?? "")
                .onSuccess { response in
                    debugPrint("native.onSuccess.response ==> \(String(describing: response.toJSON(root: "success")))")
                    self.eventSink?(response.toJSON(root: "success"))
                }.onFailure { pgErrors, exception in
                    print("Exception: \(exception)")
                    debugPrint("native.onFailure.errors ==> \(String(describing: pgErrors.toJSON(root: "failure")))")
                    self.eventSink?(pgErrors.toJSON(root: "failure"))
                }.initialize { errors in
                    debugPrint("native.initialize.errors ==> \(String(describing: errors.toJSON(root: "error")))")
                    self.eventSink?(errors.toJSON(root: "error"))
                }
            
        }
        return nil
    }
    
    public func onCancel(withArguments arguments: Any?) -> FlutterError? {
        return nil
    }
    
    
}

